### DISCUSSION TITLE
7za missing in the docker container?

### DISCUSSION THREAD

**Discussion by Tensai75** [nzbgetcom/nzbget#145] [Q&A] [answered]
I have installed the latest image (docker.io/nzbgetcom/nzbget:latest) via Portainer and it works so far. But when I encounter a 7zip archive, I get the following error: `7-Zip: Could not start /app/nzbget/7za: No such file or directory` 
When I log into the container via console, I see the following:
```
/app/nzbget # ls -lsa
total 24860
     4 drwxr-xr-x    1 1000     users         4096 Jan 16 12:57 .
     4 drwxr-xr-x    1 1000     users         4096 Jan 16 12:58 ..
     0 lrwxrwxrwx    1 1000     users           12 Jan 16 12:57 7za -> /usr/bin/7za
   224 -rw-r--r--    1 1000     users       226639 Jan 16 12:57 cacert.pem
 24616 -rwxr-xr-x    1 1000     users     25205728 Jan 16 12:57 nzbget
     4 -rw-r--r--    1 1000     users          451 Jan 16 12:57 pubkey.pem
     4 drwxr-xr-x    1 1000     users         4096 Jan 16 12:57 share
     0 lrwxrwxrwx    1 1000     users           14 Jan 16 12:57 unrar -> /usr/bin/unrar
     4 drwxr-xr-x    1 1000     users         4096 Jan 16 12:57 webui
/app/nzbget # cd /usr/bin
/usr/bin # ls -lsa
total 3288
     4 drwxr-xr-x    1 root     root          4096 Nov  9 12:44 .
     4 drwxr-xr-x    1 root     root          4096 Nov  9 12:44 ..
     0 lrwxrwxrwx    1 root     root             9 Nov  9 12:44 2to3 -> 2to3-3.11
     4 -rwxr-xr-x    1 root     root            96 Oct  4 06:22 2to3-3.11
     0 lrwxrwxrwx    1 root     root             3 Nov  9 12:44 7z -> 7zz
  1644 -rwxr-xr-x    1 root     root       1681776 Apr 20  2023 7zz
     0 lrwxrwxrwx    1 root     root            12 Sep 28 11:18 [ -> /bin/busybox
     0 lrwxrwxrwx    1 root     root            12 Sep 28 11:18 [[ -> /bin/busybox
     0 lrwxrwxrwx    1 root     root            12 Sep 28 11:18 awk -> /bin/busybox
     0 lrwxrwxrwx    1 root     root            12 Sep 28 11:18 basename -> /bin/busybox
     [...]
```
`/app/nzbget/7za` points to `/usr/bin/7za` but ther is no `7za` only a `7zz` file?

Am I missing something?

Regards, Tensai

**Comment by phnzb** [comment, marked as answer]
@Tensai75 nzbgetcom docker container is using 7z from alpine packages, and by default for fresh install SevenZipCmd is `7z`. It looks like you are migrating from a Linuxserver image with preserving the config, or you ran a latest image before the latest fixes. See [migrate instructions](https://github.com/nzbgetcom/nzbget/issues/84#issuecomment-1884846500) p.3 and fix, or stop container, remove nzbget.conf, pull latest image, and run container again.

**Reply by Tensai75** [reply]
I thought I must have overlooked something.
As a solution, I simply changed the value in the nzbget.config to `SevenZipCmd=7z`. There was no need to delete the config file (and lose my valuable settings), and after restarting it now works fine.

Many thanks, Tensai

**Comment by paul-chambers** [comment]
Sounds like something the import should check for? maybe even fix automagically.

**Reply by Tensai75** [reply]
Well, I think the docker file needs fixing.
In the docker file, a symlink is created which points `/app/nzbget/7za` to `/usr/bin/7za`. But this makes no sense, because there is no /usr/bin/7za file in the image. Probably this command was taken from the old linuxserver.io container as well?

https://github.com/nzbgetcom/nzbget/blob/fcc104dc294a905155f65e93483de4a1f40d843e/docker/Dockerfile#L37

If the line is changed to point to `/usr/bin/7z` the zip support in the nzbget.com docker image would work with legacy nzbget.config files from the old linuxserver.io container perfectly.

Regards, Tensai

**Reply by phnzb** [reply]
@Tensai75 Agreed
7za link fixed in all nzbgetcom images. When migrating, no need to change existing unpack settings. For fresh installs settings remain same (7z / unrar).
